### DISCUSSION TITLE
Point picking with panel backend

### DISCUSSION THREAD

**Discussion by ZbindenJoel** [pyvista/pyvista#2491] [Q&A] [answered]
I am building a web application, that features an interactive 3D model that also requires the point picking feature. 

Now I am trying out different plotting backends in a jupyter environment and I am especially interested in the `panel` backend, since the panel library allows for direct integration of the rendering window via their [VTK-widget](https://panel.holoviz.org/reference/panes/VTK.html). 

I succesfully enabled and used the point picking feature when using the standard plotting backend (`ipyvtklink`) but the feature seemingly stops to work when used with any of the other backends referenced in the [pyvista.Plotter.show](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.show.html) docs or when directly integrated into a panel VTK-widget.

Example code:
```python
import numpy as np
import pyvista as pv

points = np.array([[0, 0, 0], [0, 1, 0], [1, 0, 0], [0, 0, 1]], dtype=np.float32)
faces = [3, 0, 1, 2, 3, 0, 1, 3, 3, 1, 2, 3, 3, 2, 0, 3]

mesh = pv.PolyData(
    var_inp = points, 
    faces=faces,
)

# get plotter
pl = pv.Plotter(notebook=True)

# add mesh to scene
actor = pl.add_mesh(mesh)

# enable point picking
pl.enable_point_picking(use_mesh=True, pickable_window=False, show_point=True)
plot_backend = 'ipyvtklink' # "ipyvtklink", "panel", "pythreejs", "ipygany"
pl.show(jupyter_backend=plot_backend)
```

**Comment by banesullivan** [comment, marked as answer]
At this time, the `ipyvtklink` backend is the only one capable of support cell picking like this. All the other backend export the scene to a JS-based rendering framework.

You could perform the picking with VTK.js in panel but it'd be substantial work.

I'd recommend also trying out `trame` which is well built for this use case: https://kitware.github.io/trame/
